### PR TITLE
Add fstab and helpers

### PR DESCRIPTION
**Pull request by damacus**
- Add a fstab for centos-7 & fedora-latest
- Remove matchers
-

### Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: .kitchen.dokken.yml [hunk 1]
```diff
@@ -56,12 +56,16 @@ platforms:
     image: dokken/centos-7
     privileged: true
     pid_one_command: /usr/lib/systemd/systemd
+    internediate_instructions:
+      - RUN if [ ! -f /etc/fstab ];then touch /etc/fstab fi
```
Comment (tas50) [inline .kitchen.dokken.yml]: any reason not to just always touch this?

**Comment by damacus** [comment]
Because I'm a numpty?

**Comment by tas50** [comment]
There's some Chef 14 deprecation warnings in there as well

**Comment by damacus** [comment]
This is working as described now but I'm still getting an error from centos-7 not finding fstab even though the file "should" be there. Something looks like it's deleting it. 

` No such file or directory @ rb_sysopen - /etc/fstab`

**Comment by damacus** [comment]
Right this is good except the upstream "bug" in either Centos/Amazon or the syctl cookbook failing on a badly set key.

**Comment by tnguyen14** [comment]
Why are the matchers removed? What should I use instead of `create_swap_file`?

**Comment by tas50** [comment]
Matchers are autogenerated by ChefDK now. If this causes failures you need to update ChefDK to a more recent release.

**Comment by tnguyen14** [comment]
Ok. With this change, will the new matcher be `create_swap_file` or `create_swapfile`?

File: resources/file.rb [hunk 2]
```diff
@@ -49,139 +37,5 @@
...
-  end
-
-  def create_swapfile(command)
-    converge_by "create empty swapfile at #{new_resource.path}" do # ~FC054
-      shell_out!(command, timeout: new_resource.timeout)
-    end
-  end
-
-  def set_permissions
-    permissions = '600'
```
Comment (ehaselwanter) [inline resources/file.rb]: why here `600` but in the new code helper `60`

File: libraries/swap_helpers.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    end
+
+    def create_swapfile(command)
+      converge_by "create empty swapfile at #{new_resource.path}" do # ~FC005
+        shell_out!(command, timeout: new_resource.timeout)
+      end
+    end
+
+    def set_permissions
+      permissions = 60
```
Comment (ehaselwanter) [inline libraries/swap_helpers.rb]: why here `60` but in the old code `600` ?

I guess this leads to:

```
       ---- Begin output of swapon /opt/swap ----
       STDOUT:
       STDERR: swapon: /opt/swap: insecure permissions 0060, 0600 suggested.
       swapon: /opt/swap: swapon failed: Invalid argument
       ---- End output of swapon /opt/swap ----
       Ran swapon /opt/swap returned 255
```

File: .kitchen.dokken.yml [hunk 4]
```diff
@@ -56,12 +56,16 @@ platforms:
     image: dokken/centos-7
     privileged: true
     pid_one_command: /usr/lib/systemd/systemd
+    internediate_instructions:
```
Comment (vamega) [inline .kitchen.dokken.yml]: Should this be spelt `intermediate_instructions` instead?

File: .kitchen.dokken.yml [hunk 5]
```diff
@@ -56,12 +56,16 @@ platforms:
...
+      - RUN touch /etc/fstab
 
 - name: fedora-latest
   driver:
     image: dokken/fedora-latest
     privileged: true
     pid_one_command: /usr/lib/systemd/systemd
+    internediate_instructions:
```
Comment (vamega) [inline .kitchen.dokken.yml]: Should this be spelt `intermediate_instructions` instead?

File: .kitchen.dokken.yml (hunk 4 above)
Comment (damacus) [reply .kitchen.dokken.yml]: Yep, fancy making a PR?

Comment (vamega) [reply .kitchen.dokken.yml]: Coming right up.

**Comment by tnguyen14** [comment]
@tas50 do you know what the matcher that is automatically created for chefdk is?

**Pull request merged**